### DISCUSSION TITLE
Install and configuring KKP on on-premises

### DISCUSSION THREAD

**Discussion by udupicloud** [kubermatic/kubermatic#11078] [Q&A]
Hi everyone,

I am new to KKP and have basic knowledge of deployments in K8s for a year.
Please help me.

When I run to setup the KKP on on-premises. I get below given error. How do I fix it.

ceph@node3:~$ ./kubermatic-installer deploy --config examples/kubermatic.example.yaml --helm-values examples/values.example.yaml  --storageclass azure  --kubeconfig ~/.kube/config
INFO[0000] 🚀 Initializing installer…                     edition="Community Edition" version=v2.21.1
INFO[0000] 🚦 Validating the provided configuration…     
WARN[0000]    Helm values: kubermaticOperator.imagePullSecret is empty, setting to spec.imagePullSecret from KubermaticConfiguration 
INFO[0000] ✅ Provided configuration is valid.           
INFO[0000] 🚦 Validating existing installation…          
INFO[0000] ✅ Existing installation is valid.            
INFO[0000] 🛫 Deploying KKP master stack…                
INFO[0000]    💾 Deploying kubermatic-fast StorageClass… 
INFO[0000]    ✅ Success.                                
INFO[0000]    📦 Deploying nginx-ingress-controller…     
INFO[0000]       Deploying Helm chart…                  
INFO[0158]    ✅ Success.                                
INFO[0158]    📦 Deploying cert-manager…                 
INFO[0158]       Deploying Custom Resource Definitions… 
INFO[0160]       Deploying Helm chart…                  
INFO[0273]    ✅ Success.                                
INFO[0273]    📦 Deploying Dex…                          
INFO[0334]    ✅ Success.                                
INFO[0334]    📦 Deploying Kubermatic Operator…          
INFO[0334]       Deploying Custom Resource Definitions… 
INFO[0339]       Migrating UserSSHKeys…                 
INFO[0340]       Migrating Users…                       
INFO[0340]       Migrating ExternalClusters…            
INFO[0340]       Deploying Helm chart…                  
INFO[0439]    ✅ Success.                                
INFO[0439]    📝 Applying Kubermatic Configuration…      
INFO[0439]    ✅ Success.                                

ERRO[0439] ❌ Operation failed: failed to apply Kubermatic Configuration: KubermaticConfiguration.kubermatic.k8c.io "kubermatic" is invalid: spec.ingress.certificateIssuer.apiGroup: Invalid value: "null": spec.ingress.certificateIssuer.apiGroup in body must be of type string: "null". 

Currently I am deploying on 2 nodes, once things goes good will add additional and also move to cloud.

Thanks and Regards,
Mark.

**Comment by xrstf** [comment]
Can you post your KubermaticConfiguration (please remove credentials and secrets first)?
